### PR TITLE
fix: delete and Cleanup now have manifests plumbed through

### LINKED ISSUE #7456: `skaffold delete` not working @ HEAD (52c669b7113e37ee8d06f2cdd2a019572a7298b8)
Currently `skaffold delete` appears to be broken at head - 52c669b7113e37ee8d06f2cdd2a019572a7298b8

It seems `skaffold delete` currently passes 0 manifests to the `Delete` method.  This makes `skaffold delete` succeed (which is why I assume tests are passing) but not actually delete the proper resources. 

This was pointed out initially by @bskaplan who stated:
https://github.com/GoogleContainerTools/skaffold/blob/main/pkg/skaffold/runner/v2/cleanup.go#L24 is currently passing nil for the manifest list. 

Example logs for skaffold v2 (from HEAD)
```
aprindle@aprindle ~/skaffold/examples/microservices  [main]$ skaffold run
Generating tags...
 - leeroy-web -> leeroy-web:v1.38.0-70-g52c669b71
 - leeroy-app -> leeroy-app:v1.38.0-70-g52c669b71
 - base -> base:v1.38.0-70-g52c669b71
# ... run succeeds

aprindle@aprindle ~/skaffold/examples/microservices  [main]$ skaffold delete
Cleaning up...
 - No resources found

```

Logs from v1.38.0
```
aprindle@aprindle ~/skaffold/examples/microservices  [main]$ skaffold-v1.38.0 run
Generating tags...
 - leeroy-web -> leeroy-web:v1.38.0-70-g52c669b71
 - leeroy-app -> leeroy-app:v1.38.0-70-g52c669b71
 - base -> base:v1.38.0-70-g52c669b71
# ... run succeeds

aprindle@aprindle ~/skaffold/examples/microservices  [main]$ skaffold-v1.38.0 delete
Cleaning up...
 - deployment.apps "leeroy-web" deleted
 - service "leeroy-app" deleted
 - deployment.apps "leeroy-app" deleted

```

### PR DESCRIPTION
**Pull request by aaron-prindle**
Fixes #7456

### CODE REVIEW

File: cmd/skaffold/app/cmd/delete.go [hunk 1]
```diff
@@ -43,7 +52,34 @@ func NewCmdDelete() *cobra.Command {
 }
 
 func doDelete(ctx context.Context, out io.Writer) error {
-	return withRunner(ctx, out, func(r runner.Runner, _ []util.VersionedConfig) error {
-		return r.Cleanup(ctx, out, dryRun)
+	return withRunner(ctx, out, func(r runner.Runner, configs []util.VersionedConfig) error {
+		workDir, err := sutil.RealWorkDir()
```
Comment (MarlonGamez) [inline cmd/skaffold/app/cmd/delete.go:56]: can we move the added code here into a function to reuse here and in `cmd/dev.go`? Looks like it's almost the same code

**Review by MarlonGamez — CHANGES_REQUESTED** [review]
lgtm, have one request

File: cmd/skaffold/app/cmd/delete.go (hunk 1 above)
Comment (aaron-prindle) [reply cmd/skaffold/app/cmd/delete.go]: Yep, moved deduped the logic to a method in cmd/util.go and reuse it in `cmd/dev.go` and `cmd/delete.go`

**Review by MarlonGamez — APPROVED** [review]
lgtm! Thanks for making the change :)

**Pull request merged**